src/main/scala/temple/generate/database/PostgresGenerator.scala

```diff
@@ -16,14 +16,31 @@ object PostgresGenerator extends DatabaseGenerator {
   type Context = PostgresContext
 
   /** Given an expression, parse it into the Postgres format */
-  private def generateExpression(expression: Expression): String =
+  private def generateExpression(expression: Expression, index: Int)(implicit context: PostgresContext): (String, Int) =
```

Comment:
This feels like rather a lot of awkward threading of the position int through here, especially when the context is passed around too. Could the context instead expose a `nextPreparedSymbol` method? This would mean that a fresh PostgresContext would be needed for each query to be generated. This could be achieved by

1. Replacing `implicit val context` with `implicit def context` everywhere outside this class
1. ContextFactory, though for stateless database contexts this would be a trivial implementation of `class MyDBContext extends DBContextFactory with DBContext {def make = this}`

Another way to do it could be to make a `PostgresGenerator` class, with this index as a variable